R/subset_datarqst.R

```diff
@@ -37,66 +45,141 @@
 subset_datarqst <- function(file = file.choose(),
                             PopID = unique(pop_names$code),
                             Species = unique(Species_codes$Code),
-                            filter = NULL){
+                            filter = NULL,
+                            include_conflicting = FALSE,
+                            output_type = "R",
+                            return_R = FALSE,
+                            save = TRUE,
+                            test = FALSE){
+
+  if(!test){
+    readRDS(file = file)
+  }else{
+    standard_data <- test_data
+  }
 
-  standard_data <- readRDS(file = file)
 
   if(!is.null(filter)){
 
-    unique_pops <- unique(stringr::str_split(string = filter, pattern = "_")[[1]])
+    unique_pops <- unique(matrix(unlist(stringr::str_split(string = filter, pattern = "_")), ncol = 2, byrow = T)[,1])
```

Comment:
I found it slightly difficult to read this line; however, when it works it works :) 

`stringr::str_split` has an argument "simplify" that returns a matrix rather than a list, maybe that helps the readability a bit (so for instance `unique(stringr::str_split(string = filter, pattern = "_", simplify = TRUE)[,1])`).

Reply:
Great suggestion @StefanVriend, your solution returns the identical result but is much easier to read. Have implemented it :-)